assets/js/components/api-error-placeholder/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+import classNames from 'classnames';
+import {
+	Button,
+	Placeholder,
+	Spinner,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+
+class ApiErrorPlaceholder extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			retrying: false,
+		};
+		this.onRetry = this.onRetry.bind( this );
+	}
+
+	onRetry() {
+		const { onRetry } = this.props;
+
+		this.setState( { retrying: true } );
+		onRetry();
+	}
+
+	render() {
+		const { onRetry, errorMessage, className } = this.props;
+		const { retrying } = this.state;
+		return (
+			<Placeholder
+				icon={ <Gridicon icon="notice" /> }
+				label={ __( 'Sorry, an error occurred', 'woo-gutenberg-products-block' ) }
+				className={ classNames( 'wc-block-api-error', className ) }
+			>
+				<div
+					className="wc-block-error__message"
+					dangerouslySetInnerHTML={ { __html: errorMessage } }
```

Comment:
I'm worried we might forget we have `dangerouslySetInnerHTML` here and some day might be passing dangerous code in the props. :worried: 

What do you think about using `dangerouslySetInnerHTML` directly where we create the message?

Something like modifying [this line](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/679/files#diff-417cc438ee309a6b0c3cb3695ed0b216R123) to:
```JSX
error.message = (
	<span dangerouslySetInnerHTML={ { __html: __( 'The following error was returned from the API', 'woo-gutenberg-products-block' ) + ': <br/><code>' + apiError.message + '</code>' } } />
);
```

This way we can get rid of `dangerouslySetInnerHTML` in the `<ApiErrorPlaceholder />` component.

You will also have to modify the `errorMessage` propType to `PropTypes.node` so it accepts strings and JSX nodes.